utils/src/ast-grep/resolve-binding-path.ts

```diff
@@ -68,6 +68,31 @@ function resolveBindingPathRequire(node: SgNode<Js>, path: string) {
 		});
 	}
 
+	const propertyAccesses = activeNode.findAll({
+		rule: {
+			kind: "property_identifier",
+			inside: {
+				kind: "member_expression",
+			},
+		},
+	});
+
+	if (propertyAccesses.length) {
+		const pathArr = path.split(".");
+		let newPath = ["$"];
+		let i = 0;
+
+		for (; i < propertyAccesses.length; i++) {
```

Comment:
`i` will actually be scoped to the parent of `for` automatically, so extracting it like this is unnecessary, but perhaps you did it to make it more obvious that the parent scoping is necessary/intended?

Reply:
I’m not sure if I understood it correctly, but I think that if I declare it inside the for header, it’ll only be available within the for loop’s scope, as the TypeScript compiler warns in the image below.

<img width="1170" height="256" alt="Screenshot 2025-10-08 at 15 59 38" src="https://github.com/user-attachments/assets/5b03761b-71e9-42b9-a989-bb8cd5b80827" />